include/deal.II/fe/fe_interface_values.h

```diff
@@ -1662,6 +2357,351 @@ namespace FEInterfaceViews
 
 
 
+  template <int dim, int spacedim>
+  template <class InputVector, class OutputVector>
+  void
+  Scalar<dim, spacedim>::get_local_dof_values(
+    const InputVector &dof_values,
+    OutputVector &     local_dof_values) const
+  {
+    const auto &interface_dof_indices =
+      this->fe_interface->get_interface_dof_indices();
+
+    AssertDimension(interface_dof_indices.size(), local_dof_values.size());
+
+    for (unsigned int i = 0; i < interface_dof_indices.size(); ++i)
+      local_dof_values[i] = dof_values(interface_dof_indices[i]);
+  }
+
+
+
+  template <int dim, int spacedim>
+  template <class InputVector>
+  void
+  Scalar<dim, spacedim>::get_function_interface_values_from_local_dof_values(
+    const bool         here_or_there,
+    const InputVector &local_dof_values,
+    std::vector<solution_value_type<typename InputVector::value_type>> &values)
+    const
+  {
+    const auto &interface_dof_indices =
+      this->fe_interface->get_interface_dof_indices();
+
+    for (unsigned int dof_index = 0; dof_index < interface_dof_indices.size();
+         ++dof_index)
+      for (unsigned int q_index = 0; q_index < values.size(); ++q_index)
+        {
+          values[q_index] += local_dof_values[dof_index] *
```

Comment:
I don't know if what I'm about to suggest would make any real performance difference, but I note in the FEValues implementation of these types of function we set an alias `const auto &value = local_dof_values[dof_index];`  outside the quadrature loop (we reference the `value` more than once in those functions). I'd only suggest changing it if someone confirms that it would actually make a meaningful difference.

Reply:
What do you think? @tjhei

Reply:
I doubt that it matters and it is probably not the kind of optimization to do now, considering that we don't take advantage of the structure of shape functions.